test/unit/com/unifina/domain/CustomStreamIDValidatorSpec.groovy

```diff
@@ -5,33 +5,34 @@ import spock.lang.Unroll
 
 class CustomStreamIDValidatorSpec extends Specification {
 
-	private static final String MOCK_DOMAIN = "sub.my-domain.eth"
+	private static final String MOCK_ENS_DOMAIN = "sub.my-domain.eth"
 	private static final String MOCK_INTEGRATION_KEY_ADDRESS = "0xAbcdeabCDE123456789012345678901234567890"
 
 	@Unroll
 	void "validate #value"(String value, Boolean expected) {
 		def User mockUser = new User()
-		def domainValidator = {domain, creator -> ((creator == mockUser) && (domain.equals(MOCK_DOMAIN) || (domain.equals(MOCK_INTEGRATION_KEY_ADDRESS))))}
+		def domainValidator = {domain, creator -> ((creator == mockUser) && (domain.equals(MOCK_ENS_DOMAIN) || domain.equalsIgnoreCase(MOCK_INTEGRATION_KEY_ADDRESS)))}
 		def validator = new CustomStreamIDValidator(domainValidator)
 		expect:
 		validator.validate(value, mockUser) == expected
 		where:
 		value | expected
-		"sandbox/a" | true
-		"sandbox/abc/def" | true
-		"sandbox/abc/def/file.txt" | true
-		"sandbox/foo.bar/lorem.ipsum" | true
-		"sandbox/foo-bar/lorem.ipsum" | true
+		"sub.my-domain.eth/a" | true
+		"sub.my-domain.eth/abc/def" | true
+		"sub.my-domain.eth/abc/def/file.txt" | true
+		"sub.my-domain.eth/foo.bar/lorem.ipsum" | true
+		"sub.my-domain.eth/foo-bar/lorem.ipsum" | true
 		"sub.my-domain.eth/abc/def" | true
 		"0xAbcdeabCDE123456789012345678901234567890/abc/def" | true
-		"sandbox/foo-bar/lorem~ipsum" | false
-		"sandbox/abc/def/" | false
-		"sandbox/abc/def/file." | false
-		"sandbox/foo//bar" | false
+		"0xaBCDEABcde123456789012345678901234567890/abc/def" | true
+		"sub.my-domain.eth/foo-bar/lorem~ipsum" | false
+		"sub.my-domain.eth/abc/def/" | false
+		"sub.my-domain.eth/abc/def/file." | false
+		"sub.my-domain.eth/foo//bar" | false
 		"foobar.eth/abc/def" | false
+		"sub.my-domain.eth" | false
 		"0x1111111111111111111111111111111111111111/abc/def" | false
 		"foo" | false
-		"sandbox" | false
```

Comment:
Do we need a test for "0x1111111111111111111111111111111111111111"?

Reply:
No, because we already test `0x1111111111111111111111111111111111111111/abc/def` (where user doesn't own the address) and `sub.my-domain.eth` (where user owns the address, but the syntax is malformed).